source/extensions/load_balancing_policies/client_side_weighted_round_robin/client_side_weighted_round_robin_lb.cc

```diff
@@ -146,14 +161,25 @@ void ClientSideWeightedRoundRobinLoadBalancer::updateWeightsOnHosts(const HostVe
   if (!weights.empty()) {
     auto median_it = weights.begin() + weights.size() / 2;
     std::nth_element(weights.begin(), median_it, weights.end());
-    default_weight = *median_it;
+    if (weights.size() % 2 == 1) {
```

Comment:
IIUC this is a bug fix, and there should be an accompanying release note for this.

Reply:
Done